ksqldb-rest-app/src/test/java/io/confluent/ksql/api/server/LoggingHandlerTest.java

```diff
@@ -0,0 +1,90 @@
+package io.confluent.ksql.api.server;
+
+import static io.confluent.ksql.api.server.LoggingHandler.HTTP_HEADER_USER_AGENT;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import io.confluent.ksql.rest.server.KsqlRestConfig;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.net.SocketAddress;
+import io.vertx.ext.web.RoutingContext;
+import java.util.function.Consumer;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class LoggingHandlerTest {
+
+  @Mock
+  private Server server;
+  @Mock
+  private Consumer<String> logger;
+  @Mock
+  private RoutingContext routingContext;
+  @Mock
+  private KsqlRestConfig ksqlRestConfig;
+  @Mock
+  private HttpServerRequest request;
+  @Mock
+  private HttpServerResponse response;
+  @Mock
+  private SocketAddress socketAddress;
+  @Captor
+  private ArgumentCaptor<String> logStringCaptor;
+  @Captor
+  private ArgumentCaptor<Handler<AsyncResult<Void>>> endCallback;
+
+  private LoggingHandler loggingHandler;
+
+
+  @Before
+  public void setUp() {
+    when(server.getConfig()).thenReturn(ksqlRestConfig);
+    when(routingContext.response()).thenReturn(response);
+    when(routingContext.request()).thenReturn(request);
+    when(request.remoteAddress()).thenReturn(socketAddress);
+    when(ksqlRestConfig.getList(any())).thenReturn(ImmutableList.of("401"));
+    loggingHandler = new LoggingHandler(server, logger);
+  }
+
+  @Test
+  public void shouldProduceLog() {
+    when(response.getStatusCode()).thenReturn(200);
+    when(request.uri()).thenReturn("/query");
+    when(request.getHeader(HTTP_HEADER_USER_AGENT)).thenReturn("bot");
+    when(socketAddress.host()).thenReturn("123.111.222.333");
+    when(request.bytesRead()).thenReturn(3456L);
+    loggingHandler.handle(routingContext);
```

Comment:
nit: these three lines can be moved into setUp() and removed from the individual tests.

Reply:
I moved most of this to setup.  The handle line actually differs between tests.